x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/PushTopNToSource.java

```diff
@@ -204,6 +220,10 @@ private static boolean canPushDownOrders(List<Order> orders, LucenePushdownPredi
         return orders.stream().allMatch(o -> isSortableAttribute.apply(o.child(), lucenePushdownPredicates));
     }
 
+    private static boolean canPushLimit(TopNExec topn, PhysicalSettings physicalSettings) {
+        return topn.limit() instanceof Literal l && ((Number) l.value()).intValue() <= physicalSettings.luceneTopNLimit();
+    }
```

Comment:
NIT: this could be simplified with https://github.com/elastic/elasticsearch/blob/2a1176a4d904eaf47b98b2f932745cba1d6ade26/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/Foldables.java#L99-L107